includes/multi-currency/Compatibility.php

```diff
@@ -87,12 +87,41 @@ public function override_selected_currency() {
 	}
 
 	/**
-	 * Checks to see if the widgets should be hidden.
+	 * Deprecated method, please use should_disable_currency_switching.
 	 *
 	 * @return bool False if it shouldn't be hidden, true if it should.
 	 */
 	public function should_hide_widgets(): bool {
-		return apply_filters( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', false );
+		wc_deprecated_function( __FUNCTION__, '6.5.0', 'Compatibility::should_disable_currency_switching' );
+		return $this->should_disable_currency_switching();
+	}
+
+	/**
+	 * Checks to see if currency switching should be disabled, such as the widgets and the automatic geolocation switching.
+	 *
+	 * @return bool False if no, true if yes.
+	 */
+	public function should_disable_currency_switching(): bool {
+		$return = false;
+
+		/**
+		 * If the pay_for_order parameter is set, we disable currency switching.
+		 *
+		 * WooCommerce itself handles all the heavy lifting and verification on the Order Pay page, we just need to
+		 * make sure the currency switchers are not displayed. This is due to once the order is created, the currency
+		 * itself should remain static.
+		 */
+		if ( isset( $_GET['pay_for_order'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$return = true;
+		}
+
+		// If someone has hooked into the deprecated filter, throw a notice and then apply the filtering.
+		if ( has_action( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets' ) ) {
+			wc_deprecated_hook( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', '6.5.0', MultiCurrency::FILTER_PREFIX . 'should_disable_currency_switching' );
+			$return = apply_filters( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', $return );
```

Comment:
Should we directly return the outcome here, because it will also run the next filter for those who used the two filters together? If the request is coming from `should_hide_widgets`, there won't be a need to run the `should_disabld_currency_switching` filter here.